### PR TITLE
Chrusher backfiller delete

### LINKED ISSUE #110: Configurable window of "Last N Hours" of segments to keep
To minimize disk requirements on "just in case" capture nodes, we want to configure them to only keep, say, the last 24 hours of data.

This requires two features:
* Able to configure backfiller to only backfill up to N hours ago
* Some component (this could probably also be backfiller) to delete expired data.

### PR DESCRIPTION
**Pull request by chrusher**
Added the option to automatically delete hours older than the number of hours ago (or specific hour) given by `start`.

Fixes #110

### CODE REVIEW

File: backfiller/backfiller/main.py [hunk 1]
```diff
@@ -220,6 +242,53 @@ def stop_worker(self, node):
...
+			hours = [hour for hour in hours if datetime.datetime.strptime(hour, HOUR_FMT) < cutoff]
+			hours.sort()
+
+			for hour in hours:
+				path = os.path.join(self.base_dir, self.channel, quality, hour)
+				self.logger.info('Deleting {}'.format(path))
+				segments = list_local_segments(self.base_dir, self.channel, quality, hour)
+				for segment in segments:
+					try:
+						os.remove(os.path.join(path, segment))
```
Comment (ekimekim) [inline backfiller/backfiller/main.py]: we should have a metric tracking deleted segments, ideally with hour as label

File: backfiller/backfiller/main.py [hunk 2]
```diff
@@ -262,6 +332,9 @@ def run(self):
 			if self.run_once:
 				break
 
+			if self.delete_old and self.start:
+				self.delete_hours()
```
Comment (ekimekim) [inline backfiller/backfiller/main.py]: if the DB is down / get_nodes() is erroring, deletes won't occur.
this isn't a problem, but it should be noted somewhere. even just a comment here is fine.

**Comment by chrusher** [comment]
Given that deleting old hours isn't key to the operation of the backfiller, I don't want an error from deleting segments to crash the whole backfiller. Is it worth worrying about more exotic failure conditions than files/folders already being deleted or folders being non-empty?

**Comment by ekimekim** [comment]
I think what you've got covers all expected conditions.

It might make sense to wrap the delete in a general catch and log warning about it, instead of crashing.

**Pull request merged**